src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/IProjectTreeExtensions.cs

```diff
@@ -0,0 +1,15 @@
+﻿using Microsoft.VisualStudio.Shell;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS
+{
+    internal static class IProjectTreeExtensions
+    {
+        public static HierarchyId GetHierarchyId(this IProjectTree tree) =>
+            new HierarchyId(tree.IsRoot() ? VSConstants.VSITEMID_ROOT : unchecked((uint)tree.Identity.ToInt32()));
```

Comment:
Maybe my understanding is flawed, but couldn't it potentially be possible for the ID to be larger than 2^31-1? Highly unlikely, sure, but possible?